src/freenet/client/async/SplitFileFetcherStorage.java

```diff
@@ -177,7 +177,12 @@
     final int cooldownTries;
     /** Cooldown lasts this long for each key. */
     final long cooldownLength;
-    /** Only set if all segments are in cooldown. */
+    /** Separate lock for cooldown operations, which must be serialized. Must be taken *BEFORE*
+     * segment locks. DO NOT TAKE THIS LOCK WHILE HOLDING ANY OTHER LOCK! */
```

Comment:
Why can't this lock be taken while holding any other lock? It'd also be nice for this comment to include less all-caps shouting.